src/Fixer/Phpdoc/PhpdocNoEmptyReturnFixer.php

```diff
@@ -105,7 +105,7 @@ protected function applyFix(\SplFileInfo $file, Tokens $tokens)
      */
     private function fixAnnotation(DocBlock $doc, Annotation $annotation)
     {
-        if (1 === Preg::match('/@return\s+(void|null)(?!\|)/', $doc->getLine($annotation->getStart())->getContent())) {
+        if (1 === Preg::match('/@return\s+(void|null)(?!\||\[)/', $doc->getLine($annotation->getStart())->getContent())) {
```

Comment:
maybe instead of negative lookahead, let us ensure there is whitespace character after `null` ? or use DocBlock component here and rely on logic that is already there...

Reply:
There doesn't have to be a whitespace character, as discussed before on gitter. But you wanted me to fix that case as separate PR 😉

Reply:
still, I believe we shall switch to DocBlock, and if issue exist fix it there, for all fixers, and not here, for single fixer